lib/kakine/config.rb

```diff
@@ -12,25 +19,36 @@ def self.setup
 
     def self.load_config
       config_file = File.join(Dir.home, '.kakine')
-      raise '~/.kakine is missing' unless File.exists?(config_file)
+      return false unless File.exists?(config_file)
 
       config = YAML.load_file(config_file)
 
-      %w[auth_url tenant username password].each do |conf_item|
-        raise "Configuration '#{conf_item}' is missing. Check your ~/.kakine" unless config[conf_item]
-      end
-
       @@auth_url       = config['auth_url']
-      @@tenant         = config['tenant']
+      @@tenant_name    = config['tenant_name'] || config['tenant']  # for compatibility
       @@username       = config['username']
       @@password       = config['password']
       true
     end
 
+    def self.load_env
+      @@auth_url       = ENV['OS_AUTH_URL'] if ENV['OS_AUTH_URL']
+      @@tenant_name    = ENV['OS_TENANT_NAME'] if ENV['OS_TENANT_NAME']
+      @@username       = ENV['OS_USERNAME'] if ENV['OS_USERNAME']
+      @@password       = ENV['OS_PASSWORD'] if ENV['OS_PASSWORD']
+    end
+
+    def self.validate_config
+      %w[auth_url tenant_name username password].each do |conf_item|
```

Comment:
what do you mean by missing variable? The class variables accessed with class_variable_get?